projects/components/src/overlay/overlay.service.ts

```diff
@@ -46,10 +52,13 @@ export class OverlayService implements OnDestroy {
     return popover;
   }
 
-  public createModal(config: ModalOverlayConfig, injector: Injector = this.defaultInjector): PopoverRef {
+  public createModal<TData = unknown>(
+    config: ModalOverlayConfig<TData>,
+    injector: Injector = this.defaultInjector
+  ): ModalRef {
     this.activeModalPopover?.close();
 
-    const popover = this.popoverService.drawPopover({
+    const modal = this.modalService.drawPopover({
```

Comment:
If we want to put this in its own service, let's remove the method here and have callers use modal service directly.